services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -113,4 +126,33 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
                             conflictedFlow.get().getFlowId()));
         }
     }
+
+    /**
+     * Ensure switches are exists.
+     *
+     * @param requestedFlow a flow to be validated.
+     * @throws SwitchValidationException if switch not found.
+     */
+    @VisibleForTesting
+    void validateSwitches(Flow requestedFlow) throws SwitchValidationException {
+        final SwitchId sourceId = requestedFlow.getSourceSwitch();
+        final SwitchId destinationId = requestedFlow.getDestinationSwitch();
+
+        List<SwitchInfoData> switches = pathComputer.getSwitchPair(sourceId, destinationId);
+
+        boolean source = switches.stream().anyMatch(sw -> Objects.equals(sw.getSwitchId(), sourceId));
```

Comment:
Again you make 2 loops (two .anyMatch).

```
        boolean source = false;
        boolean destination = false;
        for (SwitchInfoData entity : switches) {
            source = source || sourceId.equals(entity.getSwitchId());
            destination = destination || destinationId.equals(entity.getSwitchId());
        }
```

PS You don't need so verbose error message, just list "missing" switches without naming which one is source/dest.
PPS Your current reporting scheme is missing one more option/branch - single/same switch flow - when both endpoints attached to the same switch.